Sources/LanguageServerProtocol/SupportTypes/WorkspaceSettings.swift

```diff
@@ -75,3 +80,52 @@ public struct ClangCompileCommand: Codable, Hashable {
     self.workingDirectory = workingDirectory
   }
 }
+
+/// Workspace's build scheme, represented by a list of target identifiers
+public struct BuildScheme: Codable, Hashable {
+  private enum CodingKeys: String, CodingKey {
+    case sourcekit_lsp = "sourcekit-lsp"
+    case scheme
+  }
+
+  /// all targets included in the scheme
+  public var targets: [BuildTargetIdentifier]
+
+  public init(targets: [BuildTargetIdentifier]) {
+    self.targets = targets
+  }
+  
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    guard let langExtensionDict = try? container.nestedContainer(keyedBy: CodingKeys.self, forKey: .sourcekit_lsp) else {
+        let context = DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Expected Build scheme from configuration change notification")
+        throw DecodingError.dataCorrupted(context)
+    }
+    let targetIDs = try langExtensionDict.decode([String].self, forKey: .scheme)
+    self.targets = targetIDs.map {BuildTargetIdentifier(uri: DocumentURI(string: $0))}
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    var langExtension = container.nestedContainer(keyedBy: CodingKeys.self, forKey: .sourcekit_lsp)
+    let targetIDs = self.targets.map {$0.uri.stringValue}
+    try langExtension.encode(targetIDs, forKey: .scheme)
+  }
+}
+
+public struct BuildTargetIdentifier: Codable, Hashable {
+  public var uri: DocumentURI
+
+  public init(uri: DocumentURI) {
+    self.uri = uri
+  }
+  
+  public init(from decoder: Decoder) throws {
+    let uri = DocumentURI(string: try decoder.singleValueContainer().decode(String.self))
+    self.init(uri: uri)
```

Comment:
the original type `URI` is just typealias of `DocumentURI` here if you mean the uri type here, so it's essentially the same with the original definition.
The reason of the change is because the scheme settings has to passed by lsp, this leaves `BuildTargetIdentifier` definition sits somewhere in the middle between bsp and lsp, to avoid circular reference, decided to move it into lsp.

I can change this to `String` if you mean the original type is incorrect

Reply:
he original type URI is just typealias of DocumentURI here if you mean the uri type here, so it's essentially the same with the original definition.
The reason of the change is because the scheme settings has to passed by lsp, this leaves BuildTargetIdentifier definition sits somewhere in the middle between bsp and lsp, to avoid circular reference, decided to move it into lsp.

I can change this to String if you mean the original type is incorrect

Reply:
The problem is the change from

```
{
  "uri": "file://..."
}
```

to just `"file://..."`.

The automatic Codable implementation for a struct is a keyed container (dictionary), whereas it is now a plain string.  That doesn't match the specification.